src/attribute.rs

```diff
@@ -105,43 +164,11 @@ impl Attribute {
         unsafe { from_glib_full(pango_sys::pango_attr_underline_new(underline.to_glib())) }
     }
 
-    pub fn new_variant(variant: Variant) -> Option<Attribute> {
+    /*pub fn attr_variant_new(variant: Variant) -> Option<Attribute> {
         unsafe { from_glib_full(pango_sys::pango_attr_variant_new(variant.to_glib())) }
-    }
+    }*/
 
     pub fn new_weight(weight: Weight) -> Option<Attribute> {
         unsafe { from_glib_full(pango_sys::pango_attr_weight_new(weight.to_glib())) }
     }
-
-    pub fn get_attr_class(&self) -> AttrClass {
-        unsafe { from_glib_full((*self.to_glib_none().0).klass) }
-    }
-
-    pub fn get_start_index(&self) -> u32 {
-        unsafe {
-            let stash = self.to_glib_none();
-            (*stash.0).start_index
-        }
-    }
-
-    pub fn get_end_index(&self) -> u32 {
-        unsafe {
-            let stash = self.to_glib_none();
-            (*stash.0).end_index
-        }
-    }
-
-    pub fn set_start_index(&mut self, index: u32) {
-        unsafe {
-            let stash = self.to_glib_none_mut();
-            (*stash.0).start_index = index;
-        }
-    }
-
-    pub fn set_end_index(&mut self, index: u32) {
-        unsafe {
-            let stash = self.to_glib_none_mut();
-            (*stash.0).end_index = index;
-        }
-    }
```

Comment:
Should implement `Clone` via `pango_attribute_copy()`, `Drop` via `pango_attribute_destroy()`, `PartialEq/Eq` via `pango_attribute_equal()`

Reply:
Look at the auto impl maybe?

Reply:
Ah I thought there's only the manual impl of it. Ok then